### ISSUE TITLE
[explicit-module-boundary-types] Explicit module boundary types expects private arrow function to be typed

### ISSUE THREAD

**Issue by Jamesernator** [typescript-eslint/typescript-eslint#1740]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": "error"
  }
}
```

```TS
class Foo {
  #x = () => 12;
}
```

**Expected Result**

No error.

**Actual Result**

`Missing return type on function.eslint(@typescript-eslint/explicit-module-boundary-types)`.

**Additional Info**

This doesn't affects fields like `#x = 12` or `#x = foo()`, only seen it on arrow functions.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.23.0` |
| `@typescript-eslint/parser`        | `2.23.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.11.0` |
| `npm`                              | `6.13.7` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1436

**Comment by MarkKahn** [comment]
Not just on arrow functions, I see this on non-arrow function class methods:

```ts
class A {
  #method() {
  }
}
```
